packages/world/src/systems/MachineSystem.sol

```diff
@@ -32,9 +33,18 @@ contract MachineSystem is System {
     ObjectTypeId objectTypeId = ObjectType._get(baseEntityId);
     require(objectTypeId == ForceFieldObjectID, "Invalid object type");
     EnergyData memory machineData = updateMachineEnergyLevel(baseEntityId);
+
     uint128 newEnergyLevel = machineData.energy + (uint128(numBattery) * 10);
 
-    Energy._set(baseEntityId, EnergyData({ lastUpdatedTime: uint128(block.timestamp), energy: newEnergyLevel }));
+    Energy._set(
```

Comment:
could we use the `setEnergy` util here since we're not changing drain rate or accDepletedTime?